autotest/cpp/test_ogr_geometry_stealing.cpp

```diff
@@ -0,0 +1,86 @@
+/******************************************************************************
+ *
+ * Project:  C++ Test Suite for GDAL/OGR
+ * Purpose:  Test geometry stealing from OGR feature
+ * Author:   mathieu17g (https://github.com/mathieu17g/)
+ *
```

Comment:
Please add the MIT license text (from autotest/cpp/test_ogr.cpp for example)

Reply:
Done in commit [1cf7769](https://github.com/OSGeo/gdal/pull/5196/commits/1cf7769cf09c20b492836dc209ff16898ab404ae)